starfish/image/_stack.py

```diff
@@ -960,3 +961,91 @@ def from_numpy_array(cls, array: np.ndarray) -> "ImageStack":
                     empty.set_slice({Indices.HYB: h, Indices.CH: c, Indices.Z: z}, view)
 
         return empty
+
+    def to_pixel_intensities(self, crop: Tuple[int, int, int]=(0, 0, 0)) -> IntensityTable:
+        """Generate an IntensityTable from all the pixels in the ImageStack
+
+        Parameters
+        ----------
+        crop : Tuple[int, int, int]
+            number of pixels from the image borders (z, y, x) to ignore
+
+        Returns
+        -------
+
+        """
+        zmin, ymin, xmin = crop
+        zmax = self.shape['z'] - zmin
+        ymax = self.shape['y'] - ymin
+        xmax = self.shape['x'] - xmin
+        data = self.numpy_array.transpose(2, 3, 4, 1, 0)  # (z, y, x, ch, hyb)
+        cropped_data = data[zmin:zmax, ymin:ymax, xmin:xmax, :, :]
+        intensity_data = cropped_data.reshape(-1, self.num_chs, self.num_hybs)  # (pixels, ch, hyb)
+        z = np.arange(zmin, zmax)
+        y = np.arange(ymin, ymax)
+        x = np.arange(xmin, xmax)
+
+        pixel_coordinates = pd.DataFrame(
+            data=np.array(list(product(z, y, x))),
+            columns=['z', 'y', 'x']
+        )
+        pixel_coordinates['r'] = np.full(pixel_coordinates.shape[0], fill_value=np.nan)
+
+        spot_attributes = dataframe_to_multiindex(pixel_coordinates)
+        image_size = cropped_data.shape[:3]
+
+        return IntensityTable.from_spot_data(intensity_data, spot_attributes, image_size)
+
+    @classmethod
+    def from_pixel_intensities(
+            cls, intensities: IntensityTable, assume_contiguous: bool=True
+    ) -> "ImageStack":
+        """
+
+        Parameters
+        ----------
+        intensities : IntensityTable
+            intensities to transform into an ImageStack
+        assume_contiguous : bool
+            if True, indicates that every pixel is present in the intensity_table. In this case,
+            the IntensityTable can simply be reshaped back into an ImageStack reversing
+            ImageStack.to_intensities()
+
+        Notes
+        -----
+        assume_contiguous == False is not implemented
+
+        Returns
+        -------
+        ImageStack :
+            ImageStack containing Intensity information
+
+
+        """
+        # if assume_contiguous is True, we can just reshape the original array
+        if assume_contiguous:
+
+            # reverses the process used to produce the intensity table in to_pixel_intensities
+            data = intensities.values.reshape([
+                *intensities.attrs['image_shape'],
+                intensities.sizes[Indices.CH],
+                intensities.sizes[Indices.HYB]])
+            data = data.transpose(4, 3, 0, 1, 2)
+            return ImageStack.from_numpy_array(data)
+
+        else:
+
+            raise NotImplementedError
+
+            # not implementing this right now, not clear it makes sense to.
```

Comment:
> perhaps this could be an interesting way to re-do ImageStack.show_spots?

How would this work? 

> It's unclear to me why you would want to go from pixel intensities -> an imagetack, other than for testing / simulation purposes

Great idea. Pending your explanation of the above, I'll nuke the method and move it to a testing module. 👍